src/library_html5.js

```diff
@@ -484,21 +484,34 @@ var LibraryHTML5 = {
 #if MIN_IE_VERSION != TARGET_NOT_SUPPORTED || MIN_SAFARI_VERSION <= 80000 || MIN_EDGE_VERSION <= 12 || MIN_CHROME_VERSION <= 21 // https://caniuse.com/#search=movementX
       || (e.screenY-JSEvents.previousScreenY)
 #endif
-      ;
+      ;    
 
+    var scaleX = 1;
+    var scaleY = 1;
 #if !DISABLE_DEPRECATED_FIND_EVENT_TARGET_BEHAVIOR
+    // use .canvas{X/Y}
     if (Module['canvas']) {
-      var rect = getBoundingClientRect(Module['canvas']);
-      HEAP32[idx + {{{ C_STRUCTS.EmscriptenMouseEvent.canvasX / 4 }}}] = e.clientX - rect.left;
-      HEAP32[idx + {{{ C_STRUCTS.EmscriptenMouseEvent.canvasY / 4 }}}] = e.clientY - rect.top;
+      var canvas = Module['canvas'];
+      var rect = getBoundingClientRect(canvas);
+      if (canvas.getContext && !canvas.getContext("2d")) {
```

Comment:
So here you trying to detect a 3d context?    Might be worth a comment?    To explain why we only scale for 3d contexts?

Reply:
yeah it's weird for some reason css scaling has no effect on 2D canvas mouse pos
i never had any problem with mouse pos and SDL2 while porting pygame (2D) and using scaling. Problem appeared only when changing context to 3D and i only used scaling very recently. I guess it was never tested.

Reply:
Why not use `canvas.getContext("3d")` since that seems to more directly convey the intent?

Reply:
there's only one "2D" to test, but iirc multiple 3D to tests ( webgl , webgl-experimental  ... )